aten/src/ATen/native/cuda/CuFFTPlanCache.h

```diff
@@ -69,6 +120,94 @@ static bool is_pow_of_two(int64_t x) {
   return (x & (x - 1)) == 0;
 }
 
+#ifdef __HIP_PLATFORM_HCC__
+    using cufft_size_type = int;
```

Comment:
Can we use precise types instead of int and long long int?

Reply:
This type needs to exactly match the index type used by the FFT library since it's used for pointer arguments. So no, not portably.

Reply:
Makes sense. We have the same issue with LAPACK, too, for example.